crates/client/l1-gas-price/src/worker.rs

```diff
@@ -88,18 +97,49 @@ async fn update_gas_price(
         16,
     )?;
 
-    // TODO: fetch this from the oracle
-    let eth_strk_price = 2425;
+    let response = reqwest::Client::new()
+        .get(oracle.get_fetch_url(String::from("eth"), String::from("strk")))
+        .header("x-api-key", oracle.get_api_key())
+        .send()
+        .await?;
+
+    let res_json = response.json::<OracleApiResponse>().await;
 
     let mut gas_price = gas_price.lock().await;
+
+    // We query the Oracle API for the ETH/STRK price feed
+    // If the api response is successful AND the price is within the bounds
+    // Then we update the strk_l1_gas_price and strk_l1_data_gas_price fields
+    // Otherwise we log an error and we keep the previous values
+    match res_json {
```

Comment:
Is `res_json` a `Result<serde_json::Value, _>? If not why is it named this way?

Reply:
No it's an `OracleApiResponse`. Name it accordingly

Reply:
done